general/mem_manager.cpp

```diff
@@ -143,10 +143,10 @@ struct Memory
 /// Alias class that holds the base memory region and the offset
 struct Alias
 {
-   Memory *const mem;
-   const size_t offset, bytes;
+   Memory *mem;
```

Comment:
Why should we remove all the `const`s here? Only the `mem` field within this PR seems to be updated.
Do `offset` and `bytes` should change?

Reply:
We need to update the `offset`, in case that is different for the new alias. Updating `bytes` is probably not important because we really should not use it for anything -- if we are using it somewhere, that is probably wrong. With that in mind, we probably can just remove `bytes` from the struct.

Reply:
For consistency checks it still could be useful.

Reply:
Do you mean, having `bytes` (which I removed recently) could be useful?

Reply:
Yes, it could be useful, especially if we decide to allow dangling aliases, for range comparisons with the input `bytes` of the copy calls.

Reply:
For that, I added this check when adding alias: https://github.com/mfem/mfem/blob/8e059432ab683a2a26e55ac94b8f7aa5b1de0244/general/mem_manager.cpp#L1128

and also this check when getting device alias: https://github.com/mfem/mfem/blob/8e059432ab683a2a26e55ac94b8f7aa5b1de0244/general/mem_manager.cpp#L1207

Note that it is possible to have aliases that start at the same memory address and have different lengths -- all of these are fine as long as they do not go beyond the size of the base memory address. That's what the above checks verify.